sql/src/main/java/io/crate/analyze/user/Privilege.java

```diff
@@ -62,70 +60,24 @@
     }
 
     /**
-     * Try to match a privilege at the given collection ignoring the type.
-     * Also see {@link Privilege#matchPrivilege(Collection, Type, Clazz, String)}.
+     * Builds a hash code for the given privilege attributes.
+     * <p>grantor</p> is left out by intend, this is just an information variable.
      */
-    public static boolean matchPrivilege(Collection<Privilege> privileges,
-                                         Clazz clazz,
-                                         @Nullable String ident) {
-        return matchPrivilege(privileges, null, clazz, ident);
+    public static int hashCode(State state, Type type, Clazz clazz, @Nullable String ident) {
+        return Objects.hash(state, type, clazz, ident);
     }
 
     /**
-     * Try to match a privilege at the given collection.
-     * If none is found for the current {@link Clazz}, try to find one on the upper class.
-     * If a privilege with a {@link State#DENY} state is found, false is returned.
+     * Parses the schema name out of a given table ident string.
      */
-    public static boolean matchPrivilege(Collection<Privilege> privileges,
-                                         @Nullable Type type,
-                                         Clazz clazz,
-                                         @Nullable String ident) {
-        Privilege foundPrivilege = matchPrivilegeOfClass(privileges, type, clazz, ident);
-        if (foundPrivilege == null) {
-            switch (clazz) {
-                case SCHEMA:
-                    foundPrivilege = matchPrivilegeOfClass(privileges, type, Clazz.CLUSTER, ident);
-                    break;
-                case TABLE:
-                    String schemaIdent = TableIdent.fromIndexName(ident).schema();
-                    foundPrivilege = matchPrivilegeOfClass(privileges, type, Clazz.SCHEMA, schemaIdent);
-                    if (foundPrivilege == null) {
-                        foundPrivilege = matchPrivilegeOfClass(privileges, type, Clazz.CLUSTER, null);
-                    }
-            }
-        }
-        if (foundPrivilege == null) {
-            return false;
-        }
-        switch (foundPrivilege.state()) {
-            case GRANT:
-                return true;
-            case DENY:
-            default:
-                return false;
+    public static String schemaNameOfTableIdent(String ident) {
```

Comment:
what about moving this to the Schemas class as a static method called schemaNameFromIndexName. i think for example https://github.com/crate/crate/blob/7c1133685b294dfd5108bd845242409320e31e9c/sql/src/main/java/io/crate/operation/reference/sys/shard/ShardSchemaNameExpression.java#L36 would also benefit from it